### PR TITLE
Support updates to forks of this repository

### PR DESCRIPTION
**Pull request by bbguimaraes**
When executed against a fork:
- the pruner script will not be executed, to avoid removing branches
  accidentaly (e.g. PR branches)
- a user-specific key will be used for pushes

@jupierce

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -16,22 +16,29 @@ node('openshift-build-1') {
...
-          sh '''\
-virtualenv env/
-. env/bin/activate
+        stage('run') {
+          final url = sh(
+            returnStdout: true,
+            script: 'git config remote.origin.url')
+          def prune = 'true', key = 'openshift-bot'
+          if(url.trim() != 'git@github.com:openshift/aos-cd-jobs.git') {
+            prune = ''
```
Comment (jupierce) [inline Jenkinsfile]: 'false'? Seems odd to mix 'true' and empty string.

File: Jenkinsfile [hunk 2]
```diff
@@ -16,22 +16,29 @@ node('openshift-build-1') {
...
+            key = "${(url =~ /.*:([^\/]+)/)[0][1]}-aos-cd-bot"
```
Comment (jupierce) [inline Jenkinsfile]: Can we add a check at the top of this job.. if (url startsWith "git" == false ) then error(please use an ssh url)?
I know the job won't work without it, but a clean error will save time diagnosing.

File: Jenkinsfile (hunk 1 above)
Comment (bbguimaraes) [reply Jenkinsfile]: That's an old, shell style boolean.  But I agree it's not necessary here.
Changed to something better.

File: Jenkinsfile (hunk 2 above)
Comment (bbguimaraes) [reply Jenkinsfile]: Changed, but I could not find a way to extract the git remote url from jenkins,
so the check has to be done after cloning.

**Pull request merged**